plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergParquetConnectorTest.java

```diff
@@ -73,4 +74,15 @@ protected Session withSmallRowGroups(Session session)
                 .setCatalogSessionProperty("iceberg", "parquet_writer_batch_size", "10")
                 .build();
     }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        switch ("%s -> %s".formatted(setup.sourceColumnType(), setup.newColumnType())) {
+            case "row(x integer) -> row(y integer)":
+                // TODO https://github.com/trinodb/trino/issues/15822 The connector returns incorrect NULL when a field in row type doesn't exist in Parquet files
+                return Optional.of(setup.withNewValueLiteral("NULL"));
```

Comment:
Single-field row types aren't very common.
Let's make sure we have a test for renaming some, but not all fields in a `row` (eg have a row with two fields).

(We can keep this one too)

Reply:
Sent https://github.com/trinodb/trino/pull/15957